pkg/virt-controller/watch/vmi/lifecycle.go

```diff
@@ -569,42 +570,69 @@ func prepareVMIPatch(oldVMI, newVMI *virtv1.VirtualMachineInstance) *patch.Patch
 	return patchSet
 }
 
-// These "dynamic" labels are Pod labels which may diverge from the VMI over time that we want to keep in sync.
-func (c *Controller) syncDynamicLabelsToPod(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
+// These "dynamic" annotations/labels are Pod annotations/labels which may diverge from the VMI over time that we want to keep in sync.
+func (c *Controller) syncDynamicAnnotationsAndLabelsToPod(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
 	patchSet := patch.New()
-	dynamicLabels := []string{
-		virtv1.NodeNameLabel,
-		virtv1.OutdatedLauncherImageLabel,
-	}
 	podMeta := pod.ObjectMeta.DeepCopy()
-	if podMeta.Labels == nil {
-		podMeta.Labels = map[string]string{}
-	}
-	changed := false
-	for _, key := range dynamicLabels {
-		vmiVal, vmiLabelExists := vmi.Labels[key]
-		podVal, podLabelExists := podMeta.Labels[key]
-		if vmiLabelExists == podLabelExists && vmiVal == podVal {
+	type scope struct {
```

Comment:
This seems overkill imho, we just need to handle annotations and labels.

Reply:
done